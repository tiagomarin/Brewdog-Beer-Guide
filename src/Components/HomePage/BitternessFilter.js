// import React from 'react';

// const BitternessFilter = (props) => {
//   const {setBitterness} = props
//   return (
//     <select
//       className="filter-menu"
//       name="filter"
//       onChange={(e) => {
//         setBitterness(e.target.value);
//       }}
//       required
//     >
//       <option value="">filter by bitterness</option>
//       <option value="Very low">very low</option>
//       <option value="Low">Low</option>
//       <option value="Medium">Medium</option>
//       <option value="High">High</option>
//       <option value="Very high">Very high</option>
//       <option value="Ultra high">Ultra high</option>
//     </select>
//   )
// }

// export default BitternessFilter;
