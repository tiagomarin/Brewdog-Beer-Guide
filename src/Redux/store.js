import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import beersSlice from './beersSlice';

const store = configureStore({
  reducer: {
    beers: beersSlice,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
