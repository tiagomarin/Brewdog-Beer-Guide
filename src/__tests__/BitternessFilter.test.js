import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import store from '../Redux/store';
import BitternessFilter from '../Components/HomePage/BitternessFilter';

describe('Bitterness Filter', () => {
  it('renders BitternessFilter Component', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <BitternessFilter />
          </Router>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
