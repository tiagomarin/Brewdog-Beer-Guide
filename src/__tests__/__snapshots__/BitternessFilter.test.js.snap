// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bitterness Filter renders BitternessFilter Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="filter-wrapper "
      >
        <select
          class="filter-menu"
          name="filter"
        >
          <option
            value=""
          >
            All Beers
          </option>
          <option
            value="All Beers"
          >
            All Beers
          </option>
          <option
            value="Very low"
          >
            Very low
          </option>
          <option
            value="Low"
          >
            Low
          </option>
          <option
            value="Medium"
          >
            Medium
          </option>
          <option
            value="High"
          >
            High
          </option>
          <option
            value="Very high"
          >
            Very high
          </option>
          <option
            value="Ultra high"
          >
            Ultra high
          </option>
        </select>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="filter-wrapper "
    >
      <select
        class="filter-menu"
        name="filter"
      >
        <option
          value=""
        >
          All Beers
        </option>
        <option
          value="All Beers"
        >
          All Beers
        </option>
        <option
          value="Very low"
        >
          Very low
        </option>
        <option
          value="Low"
        >
          Low
        </option>
        <option
          value="Medium"
        >
          Medium
        </option>
        <option
          value="High"
        >
          High
        </option>
        <option
          value="Very high"
        >
          Very high
        </option>
        <option
          value="Ultra high"
        >
          Ultra high
        </option>
      </select>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
